clash-lib/src/Clash/Rewrite/WorkFree.hs

```diff
@@ -155,7 +155,18 @@ isWorkFreeClockOrResetOrEnable tcm e =
   if isClockOrReset tcm eTy || isEnable tcm eTy then
     case collectArgs e of
       (Prim p,_) -> Just (primName p == Text.showt 'removedArg)
-      (Var _, []) -> Just True
+      -- Only local variables with a clock type are work-free. When it is a global
+      -- variable, it is probably backed by a clock generator, which is definitely
+      -- not work-free.
```

Comment:
The local variable distinction feel arbitrary to me.

Given:
```haskell
clk = clockGen

clkGlobal = clk

f =
 let
   clkLocal = clk
   clkA = clkLocal
   clkB = clkGlobal
 in [...]
```

As I understand it, you're saying here it is ok for bindConstantVar to replace `clkA` inside of `[...]` with `clkLocal`, but not `clkB` with `clkGlobal`.
Is that what you're saying?
And if so, why?
They seem equivalent to me.

Reply:
Yes, that's what I'm saying. Inlining `clkA` everywhere will not make the circuit `f` any larger (perform work). Inlining `clkB` everywhere will make the circuit `f` larger, because it will have duplicated calls to what will ultimately be `clockGen`.